simulation_parameters/SimulationParameters.cs

```diff
@@ -314,7 +314,7 @@ private static string ReadJSONFile(string path)
         return result;
     }
 
-    private Dictionary<string, T> LoadRegistry<T>(string path, JsonConverter[] extraConverters = null)
+    private Dictionary<string, T> LoadRegistry<T>(string path, JsonConverter[]? extraConverters = null)
```

Comment:
Would empty lists work here? This seems like a case for a non-nullable parameters.

Also applies to methods below.

Reply:
I would guess it would, but I had over 2000 warnings to fix so I couldn't spend any time thinking about clear cut cases like this where the parameter is obviously meant to be nullable as `null` was specified as the default value.